tidy3d/components/heat/simulation.py

```diff
@@ -222,6 +271,26 @@ def names_exist_grid_spec(cls, val, values):
 
         return val
 
+    @pd.validator("grid_spec", always=True)
+    def warn_if_minimal_mesh_size_override(cls, val, values):
+        """Warn if minimal mesh size limit overrides desired mesh size."""
+
+        max_size = np.max(values.get("size"))
+        min_dl = val.relative_min_dl * max_size
+
+        if isinstance(val, UniformUnstructuredGrid):
+            desired_min_dl = val.dl
+        if isinstance(val, DistanceUnstructuredGrid):
+            desired_min_dl = min(val.dl_interface, val.dl_bulk)
+
+        if desired_min_dl < min_dl:
+            log.warning(
+                f"The resulting limit for minimal mesh size from parameter 'relative_min_dl={val.relative_min_dl}' is {min_dl}, while provided mesh size in 'grid_spec' is {desired_min_dl}. "
+                "Consider lowering parameter 'relative_min_dl' if a finer grid is required."
+            )
+
```

Comment:
It would still be nice to compare this against the smallest geometry feature, but there is no sure way yet on the frontend to determine that as that might come from intersection of geometry objects. We should try to fix this eventually